src/Checker.elm

```diff
@@ -0,0 +1,62 @@
+module Checker exposing (getIssues, getIssuesJSON, getIssuesString)
+
+import Configuration
+import Constraints
+import Encoders
+import Issue
+import Json.Encode
+import Models
+
+
+getIssues : Models.ParsedModule -> Configuration.Model -> List Issue.Issue
+getIssues parsed config =
+    let
+        ignoredChecks =
+            config.excludedChecks
+
+        checkAll =
+            config.checkAllDefinitions
+
+        topLevelLine =
+            parsed.topLevelComment
+                |> Maybe.map (\( range, _ ) -> range.start.row)
+                |> Maybe.withDefault 1
+
+        topLevelCommentIssues =
+            Constraints.getViolationsTopLevel parsed.topLevelComment ignoredChecks
+                |> (\violated -> Issue.fromViolationsAndTrigger violated (Issue.TopLevel parsed.topLevelComment))
+                |> (\issue -> [ ( topLevelLine, issue ) ])
+
+        otherCommentsIssues =
+            parsed.otherComments
+                |> List.map (\comment -> ( comment, Constraints.getViolationsDangling comment ignoredChecks ))
+                |> List.map (\( comment, violated ) -> ( comment, Issue.fromViolationsAndTrigger violated (Issue.Dangling comment) ))
+                |> List.map (\( ( range, _ ), issue ) -> ( range.start.row, issue ))
+
+        entityIssues =
+            parsed.entities
+                |> List.map (\ent -> ( ent, Constraints.getViolationsEntity ent checkAll ignoredChecks ))
+                |> List.map (\( ent, viol ) -> ( ent.range.start.row, Issue.fromViolationsAndTrigger viol (Issue.Entity ent) ))
+    in
+    List.concat [ topLevelCommentIssues, otherCommentsIssues, entityIssues ]
+        |> List.sortBy (\( row, _ ) -> row)
+        |> List.map (\( _, iss ) -> iss)
+        |> List.filterMap identity
+
+
+{-| Returns the string representation of the failed checks.
```

Comment:
I find it a bit shorter to start with the action:
Represents the failed checks as strings?